### PR TITLE
gitignore: Ignore GHC-generated files

### PR DESCRIPTION
**Pull request by bgamari**
Ignore files generated by the GHC build system.

### CODE REVIEW

**Comment by RyanGlScott** [comment]
Do we need to add `dist-boot` and `dist-install` as well?

Also, this needs to be rebased on top of the latest `master`.

File: .gitignore [hunk 1]
```diff
@@ -30,3 +30,9 @@ cabal.project.local
 cabal.project.local~
 .HTF/
 .ghc.environment.*
+
+# GHC
+dist-install
+dist-boot
```
Comment (RyanGlScott) [inline .gitignore]: This shouldn't be necessary. (Sorry for suggesting otherwise earlier!)

**Comment by RyanGlScott** [comment]
This still doesn't appear to be based on top of ed98b644973430e26b93a4d12f75ad963c7afd20.

**Comment by bgamari** [comment]
Fixed.

**Pull request merged**